src/configure/mod.rs

```diff
@@ -112,21 +104,47 @@ pub unsafe fn dc_job_do_DC_JOB_CONFIGURE_IMAP(context: &Context, _job: &Job) {
             let s_a = context.running_state.clone();
             let s = s_a.read().unwrap();
 
-            if !s.shall_stop_ongoing {
-                progress!(context, 1);
-
-                let mut param = dc_loginparam_read(context, &context.sql, "");
-                if param.addr.is_empty() {
-                    error!(context, 0, "Please enter an email address.",);
-                } else {
-                    let ok_to_continue0;
-                    if 0 != param.server_flags & 0x2 {
-                        // the used oauth2 addr may differ, check this.
-                        // if dc_get_oauth2_addr() is not available in the oauth2 implementation,
-                        // just use the given one.
-                        if s.shall_stop_ongoing {
-                            ok_to_continue0 = false;
-                        } else {
+            // Variables that are shared between steps:
+            let mut param: dc_loginparam_t = dc_loginparam_read(context, &context.sql, "");
+            // need all vars here to be mutable because rust thinks the same step could be called multiple times
+            // and also initilize, because otherwise rust thinks it's used while unitilized, even if thats not the case as the loop goes only forward
+            let mut param_domain = "undefined.undefined".to_owned();
+            let mut param_addr_urlencoded: String =
+                "Internal Error: this value should never be used".to_owned();
+            let mut keep_flags = std::i32::MAX;
+
+            const STEP_3_INDEX: u8 = 13;
+            let mut step_counter: u8 = 0;
+            loop {
+                step_counter = step_counter + 1;
+
+                if s.shall_stop_ongoing {
+                    // if step_counter == 3 {
+                    //     // Skip Autoconfig? (ok_to_continue 7)
```

Comment:
Well, need to resolve these comments.

Reply:
I vote for deleting them, because that strange behavior isn't in the C core as far as I can tell, but lets ask @r10s about this.

Reply:
@VP- did a PR were its removed, additionally I can't spot this behavior in the c code so I guess its safe to assume that it doesn't belong here.

Reply:
This problem with `ok_to_continue_7` is exactly what breaks manual account configuration, and is the reason for my PR (#415).

The `while`/`match` combination is still a `goto` hack. Rust has the `return` statement and the `?` operator, either of which can be used for early exit. Both also allow moving the check back into the `progress` macro. `?` even allows the macro to become a normal function. I used `return` in my PR, but `?` (and the closure returning an `Option` or `Result`) would be fine with me as well.

As for how to continue, I think manually refactoring this PR (since it includes other changes, too) would be easier than trying to merge the two PRs somehow.

Reply:
what about merging this pr first and then make another one where we move it back in the macro?

Reply:
Fine with me. I'll re-apply my changes once this PR is merged.